[ENH] Refactor statusbar, prettify

irgolic opened the pull request:
(edited 11th Feb)
##### Description of changes
- Played around with content margins
- Elide warning/error message
- Fix in/out message size
- Remove in/out message size grip
- In/out messages always have a popup
- Remove first line from popup/tooltip

Before:

![statusbar-before](https://user-images.githubusercontent.com/24586651/74244982-31c81880-4cda-11ea-8b04-4ae27c43dfbb.gif)

After:

![statusbar-after](https://user-images.githubusercontent.com/24586651/74244993-34c30900-4cda-11ea-8557-35ec7d546493.gif)

##### Includes
- [X] Code changes
- [ ] Tests
- [ ] Documentation

ales-erjavec commented on orangewidget/utils/messagewidget.py:
```diff
@@ -541,6 +541,8 @@ def __update(self):
             fulltext = ""
         self.__fulltext = fulltext
         self.setToolTip(self.__styled(self.__defaultStyleSheet, fulltext))
+        self.setVisible(True)
```
Restore the visible state as it was at entry (use `not isHidden()` to get the correct value to restore).

ales-erjavec replied:
Still, this seems wrong.

I think `self.layout().activate()` below is the confounding factor in the jittering update.
I would suggest:
1) removing that line
2) left align the icon widget so it preserves its absolute position within MessageWidget event when it's wider then necessary.

```diff
diff --git a/orangewidget/utils/messagewidget.py b/orangewidget/utils/messagewidget.py
index 4ea6f2cf8..7b3f624eb 100644
--- a/orangewidget/utils/messagewidget.py
+++ b/orangewidget/utils/messagewidget.py
@@ -356,7 +356,7 @@ class MessagesWidget(QWidget):
         self.setLayout(QHBoxLayout())
         self.layout().setContentsMargins(2, 1, 2, 1)
         self.layout().setSpacing(0)
-        self.layout().addWidget(self.__iconwidget)
+        self.layout().addWidget(self.__iconwidget, alignment=Qt.AlignLeft)
         self.layout().addSpacing(4)
         self.layout().addWidget(self.__textlabel)
         self.layout().addWidget(self.__popupicon)
@@ -552,7 +552,6 @@ class MessagesWidget(QWidget):
         else:
             self.__popuptext = fulltext
         self.__popupicon.setVisible(bool(self.__popuptext))
-        self.layout().activate()
 
     def mousePressEvent(self, event):
         if event.button() == Qt.LeftButton:
```

irgolic replied:
What role does `self.layout().activate()` play here? Left-align solves the jittering with or without `.activate()`. I'll throw it out anyway, it's probably called implicitly at some point.

irgolic replied:
Scratch that, I'm leaving it in, else this might happen: 
![Screenshot 2020-01-14 at 14 14 44](https://user-images.githubusercontent.com/24586651/72347351-dd784b80-36cf-11ea-819f-574301acef07.png)

thocevar commented:
Before and after on Win 10.

![image](https://user-images.githubusercontent.com/919223/72339884-6e96f480-36c7-11ea-9a3b-550e5fbbd745.png)

ajdapretnar commented:
Would bigger spaces lead to problems with smaller widgets? Say Distances + some Warning?

markotoplak commented:
Before and after on linux with dwm.

![before](https://user-images.githubusercontent.com/552182/72342247-583f6780-36cc-11ea-86b4-739a5f9f0383.png)
![after](https://user-images.githubusercontent.com/552182/72342256-5e354880-36cc-11ea-8a55-fb35e57468f7.png)

markotoplak commented:
Current version on linux with dwm:
![newest](https://user-images.githubusercontent.com/552182/72348687-37324300-36db-11ea-9fcf-9678638b417c.png)

irgolic commented:
Thanks @thocevar, @markotoplak! Looks good to me.

@ajdapretnar This happens. I'd assume this isn't the only instance of a warning being cut off, even without the extra margins. I'm thinking, let's display the warning message if there's space in the status bar, otherwise just display the icon, storing the message in its tooltip.
![Screenshot 2020-01-14 at 14 40 15](https://user-images.githubusercontent.com/24586651/72348941-6b096a80-36d3-11ea-9579-425ce8dc8b08.png)

ajdapretnar commented:
> let's display the warning message if there's space in the status bar, otherwise just display the icon, storing the message in its tooltip

Agree.

irgolic commented:
@ajdapretnar I've opted to elide the text rather than hide it completely.

In/out messages are now fixed in size. I've removed the popup icon – it looks like a grip rather than a hoverable/clickable item. The messages already change their background on, making it obviously interactable. They now also show a popup on no input/output. 

Before:

![statusbar-before](https://user-images.githubusercontent.com/24586651/74242026-14dd1680-4cd5-11ea-8c00-3f4d5ec088e1.gif)

After:

![statusbar-after](https://user-images.githubusercontent.com/24586651/74242033-17d80700-4cd5-11ea-8380-a6a98f22f3fc.gif)

Before merging someone should check if the status bar and its tooltips/popups look drastically different on Windows/Linux (it'll probably be fine).

markotoplak commented:
Looks good on Linux.

ales-erjavec commented on orangewidget/widget.py:
```diff
@@ -638,21 +662,28 @@ def __info(self):
-                sb.addWidget(in_msg)
-                sb.addWidget(out_msg)
+                buttons = sb.findChild(QWidget, "buttons")
+                if buttons.layout().count() != 0:
```
`buttons` can be None here if `want_main_layout` is False (the creation of the `buttons` should probably be moved to `statusBar` though).
